config/application.rb

```diff
@@ -43,5 +43,12 @@ class Application < Rails::Application
     config.middleware.use ActionDispatch::Flash
     config.middleware.use ActionDispatch::Cookies
     config.middleware.use ActionDispatch::Session::CookieStore
+
+    config.before_configuration do
+      env_file = File.join(Rails.root, 'config', 'local_env.yml')
+      YAML.load(File.open(env_file)).each do |key, value|
+        ENV[key.to_s] = value
+      end if File.exists?(env_file)
+    end
```

Comment:
Are you sure this is needed? Also, why are you using a YAML file to set env vars?

Reply:
I saw [this](https://railsapps.github.io/rails-environment-variables.html) solution for the ENV variables, "Option Three: Use a local_env.yml File", and it says that we need to have that because "The code opens the config/local_env.yml file, reads each key/value pair, and sets environment variables."

Reply:
I don't think you need a gem or even this... Wasn't this working previously? In the other PR you already accessed env vars configured in the `.env` file, right?

Reply:
Yes, because I had that code. And I ran the Backend first and then I went to the console, so that's why I had the results from the ENV's file. If I comment that code and go to the console I'll get this:

![print_](https://user-images.githubusercontent.com/55494046/184181881-5ab96883-0223-463c-8793-96dea356cfb0.png)

Reply:
Have you sourced the file?

Reply:
Probably not, how do I do that?

Reply:
Without the server or console running (or if they are running you need to restart later) and do `source .env.dev` or whatever the env file you have (not the yaml one).

Reply:
Or you could use something like [`direnv`](https://direnv.net/) to automatically set the environment variables in that folder

Reply:
I commented out that yaml code and created the files .env and .env.dev  (inside the config folder) and I ran `source .env` and `source .env.dev`, and I also tried in the console but didn't work (print 1). So I tried the same ideia with those files in the root folder but also didn't work (print 2). 

![Screenshot from 2022-08-12 10-20-50](https://user-images.githubusercontent.com/55494046/184326531-634f48f9-f25b-4715-999f-5c21b56c94cb.png)

![Screenshot from 2022-08-12 10-22-39](https://user-images.githubusercontent.com/55494046/184326585-9a3343c8-4b46-47cc-a867-ad4f1cc8db65.png)

I found [this](https://github.com/bkeepers/dotenv) information about ENV variables, but it requires installing a gem. Is this the right way to setup the ENV variables?

Reply:
How did you setup variables in the `.env` or `.env.dev`?  You probably need to add an `export` before the environment variable name, like `export NAME="name"`

Reply:
I just noticed your idea about the `direnv` after sending the message, but I'll look into it

Reply:
I forgot the export, I will add the export and try again

Reply:
Adding the export worked on the console, but when I run the app to test it, it's still not working. Do I need to add something like `require("dotenv").config()` at the top of the files where I want to use the ENV variables?

Reply:
I don't think you need to have the `dotenv` gem, Rails usually takes care of loading environment variables automatically. In fact, the REACT_APP_BACKEND_PATH variable is already being loaded